cmd/template/cluster/provider/aws.go

```diff
@@ -76,3 +196,97 @@ func WriteAWSTemplate(out io.Writer, config ClusterCRsConfig) error {
 
 	return nil
 }
+
+func getCAPAClusterTemplate(config ClusterCRsConfig) (client.Template, error) {
+	var err error
+
+	c, err := client.New("")
+	if err != nil {
+		return nil, err
+	}
+
+	templateOptions := client.GetClusterTemplateOptions{
+		ClusterName:       config.ClusterID,
+		TargetNamespace:   key.OrganizationNamespaceFromName(config.Owner),
+		KubernetesVersion: "v1.19.9",
+		ProviderRepositorySource: &client.ProviderRepositorySourceOptions{
+			InfrastructureProvider: "aws:v0.6.6",
+			Flavor:                 "machinepool",
+		},
+	}
+	os.Setenv("AWS_SUBNET", "")
+	os.Setenv("AWS_CONTROL_PLANE_MACHINE_TYPE", "")
+	os.Setenv("AWS_REGION", "")
+	os.Setenv("AWS_SSH_KEY_NAME", "")
+
+	if replicas := int64(len(config.ControlPlaneAZ)); replicas > 0 {
+		templateOptions.ControlPlaneMachineCount = &replicas
+	}
+
+	clusterTemplate, err := c.GetClusterTemplate(templateOptions)
+	if err != nil {
+		return nil, err
+	}
+	return clusterTemplate, nil
+}
+
+func newAWSClusterFromUnstructured(config ClusterCRsConfig, o unstructured.Unstructured) (*capav1alpha3.AWSCluster, error) {
+	var awscluster capav1alpha3.AWSCluster
+	{
+		err := runtime.DefaultUnstructuredConverter.
+			FromUnstructured(o.Object, &awscluster)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		awscluster.Spec.IdentityRef = &capav1alpha3.AWSIdentityReference{
+			Name: config.ClusterID,
+			Kind: capav1alpha3.ClusterRoleIdentityKind}
+
+		if config.ControlPlaneAZ != nil {
+			for _, az := range config.ControlPlaneAZ {
+				privateSubnet := capav1alpha3.SubnetSpec{AvailabilityZone: az, IsPublic: false}
+				publicSubnet := capav1alpha3.SubnetSpec{AvailabilityZone: az, IsPublic: true}
+				awscluster.Spec.NetworkSpec.Subnets = append(awscluster.Spec.NetworkSpec.Subnets, &privateSubnet, &publicSubnet)
+			}
+		}
```

Comment:
There's no need for the `nil` check. `for ... range ...` already takes that into account

```suggestion
		for _, az := range config.ControlPlaneAZ {
			privateSubnet := capav1alpha3.SubnetSpec{AvailabilityZone: az, IsPublic: false}
			publicSubnet := capav1alpha3.SubnetSpec{AvailabilityZone: az, IsPublic: true}
			awscluster.Spec.NetworkSpec.Subnets = append(awscluster.Spec.NetworkSpec.Subnets, &privateSubnet, &publicSubnet)
		}
```